libs/utils/utils.js

```diff
@@ -650,8 +650,15 @@ export function convertStageLinks({ anchors, config, hostname }) {
       .find((domain) => a.href.includes(domain));
     if (!matchedDomain) return;
     a.href = a.href.replace(a.hostname, domainsMap[matchedDomain] === 'origin'
-      ? hostname
-      : domainsMap[matchedDomain]);
+      ? hostname.to || hostname
+      : domainsMap[matchedDomain].to || domainsMap[matchedDomain]);
```

Comment:
We should use a constant for `domainsMap[matchedDomain]` and also add the optional chaining operator to make sure things don't break.

Reply:
Added a constant for `domainsMap[matchedDomain]` 👍
The optional chaining operator is not necessary as we do an early return in [here](https://github.com/adobecom/milo/pull/2971/files#diff-28218f239d8d9c4d41e1924854440b50964619e56058fdd0ae243ff5ff6c616dR651) if `matchedDomain` does not exist as a key of `domainsMap`.